redash/query_runner/big_query.py

```diff
@@ -305,19 +305,67 @@ def _list_datasets(self):
             for project_id in projects
         ])
 
-        # Submit the query to BigQuery.
-        resp = service.jobs().query(projectId=primary_project_id, body={
-            'useLegacySql':False,
-            'query': query
-        }).execute()
+        rows = self._get_results(query)
 
         # Format results as
         # [(project_id, dataset_id)]
-        return map(
+        data = map(
             lambda row: (row['f'][0]['v'], row['f'][1]['v']),
-            resp.get("rows", [])
+            rows
         )
 
+        return data
+
+    def _get_results(self, query):
```

Comment:
please add some general description to why you're doing this, so it will clear why it's needed and relevant things to note about how.